packages/components/src/button/test/index.tsx

```diff
@@ -258,6 +258,11 @@ describe( 'Button', () => {
 					description: 'Description text',
 				} )
 			).toBeVisible();
+			expect(
+				screen.getByRole( 'tooltip', {
+					hidden: true,
+				} )
+			).toHaveTextContent( 'Description text' );
```

Comment:
Good point @ciampo! I guess then we'll need to trigger a hover, and will need to add a `waitFor()` so the assertion is called again when the tooltip appears. Something like:

```suggestion
			const user = userEvent.setup();
			await user.hover(
				screen.getByRole( 'button', {
					description: 'Description text',
				} )
			);

			await waitFor( () => {
				expect(
					screen.getByRole( 'tooltip', {
						hidden: true,
					} )
				).toHaveTextContent( 'Description text' );
			} );
```

Reply:
The other tooltip tests used `tab` so I followed that, which lets us get around using `waitFor()`:

https://github.com/WordPress/gutenberg/blob/ea90b505595a89fc4c522f388ffd7fbfbf922343/packages/components/src/button/test/index.tsx#L258-L272

But if `hover` is preferable, we could use `findByRole` instead: 

```tsx
const button = screen.getByRole( 'button', {
	description: 'Description text',
} );

expect( button ).toBeVisible();

await user.hover( button );

expect(
	await screen.findByRole( 'tooltip', {
		name: 'Description text',
	} )
).toBeVisible();
```

Not sure if [this info](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-waitfor-to-wait-for-elements-that-can-be-queried-with-find) is still relevant, but it seems like `find*` is preferable over `waitFor()`.

Reply:
I'm personally ok with both `tab` and `hover`, and I also prefer using `findBy` instead of `waitFor`.

I guess we can keep the code as-is and wrap up the PR :)

Reply:
I just quickly drafted this as an example, obviously, you can use whatever makes sense the most. In any case you can totally go with `tab` and `findBy` if you wish, I think this is a minor stylistic choice that wouldn't affect the quality of the test anyway.